plugin/src/main/cpp/gdnative/godot_arcore.cpp

```diff
@@ -1,34 +1,226 @@
+		godot::register_class<ARCoreInterface>();
+		godot::register_class<ARCoreConfig>();
+	}
 }
 
 void GDN_EXPORT godot_arcore_nativescript_terminate(void *handle) {
-  godot::Godot::nativescript_terminate(handle);
+	godot::Godot::print("godot_arcore_nativescript_terminate");
+	godot::Godot::nativescript_terminate(handle);
 }
 
 void GDN_EXPORT godot_arcore_gdnative_terminate(godot_gdnative_terminate_options *options) {
-  godot::Godot::gdnative_terminate(options);
+	godot::Godot::print("godot_arcore_gdnative_terminate");
+	godot::Godot::gdnative_terminate(options);
+}
+
+////////////////////////////////////////////////////////////////////////
+// !BAS! Implementing this here for now so we can get it to work
+// but this should move into godot-cpp at some point
+
+void *godot_arvr_constructor(godot_object *p_instance) {
+	// we need to return the pointer to the wrapper object for our instance here
+
+	ARCoreInterface *data = godot::detail::get_wrapper<ARCoreInterface>(p_instance);
+	return data;
+}
+
+void godot_arvr_destructor(void *p_data) {
+	// we can ignore this here
+}
+
+godot_string godot_arvr_get_name(const void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	godot::String arvr_name = data->get_name();
+	godot_string ret_name;
+	api->godot_string_new_copy(&ret_name, (godot_string *)&arvr_name);
+
+	return ret_name;
 }
+
+godot_int godot_arvr_get_capabilities(const void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	return (godot_int) data->get_capabilities();
+};
+
+godot_bool godot_arvr_get_anchor_detection_is_enabled(const void *p_data) {
+	godot_bool ret;
+
+	ret = false; // does not apply here
+
+	return ret;
+};
+
+void godot_arvr_set_anchor_detection_is_enabled(void *p_data, bool p_enable){
+	// we ignore this, not supported in this interface!
+};
+
+godot_bool godot_arvr_is_stereo(const void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	return (godot_bool) data->is_stereo();
+};
+
+godot_bool godot_arvr_is_initialized(const void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	return (godot_bool) data->is_initialized();
+};
+
+godot_bool godot_arvr_initialize(void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	return (godot_bool) data->initialize();
+};
+
+void godot_arvr_uninitialize(void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	data->uninitialize();
+};
+
+godot_vector2 godot_arvr_get_render_targetsize(const void *p_data) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	Size2 arvr_size = data->get_render_targetsize();
+
+	godot_vector2 ret_size;
+	api->godot_vector2_new(&ret_size, arvr_size.x, arvr_size.y);
+
+	return ret_size;
+};
+
+godot_transform godot_arvr_get_transform_for_eye(void *p_data, godot_int p_eye, godot_transform *p_cam_transform) {
+	ARCoreInterface *data = (ARCoreInterface *)p_data;
+
+	godot::Transform *cam_transform = (godot::Transform *)&p_cam_transform;
```

Comment:
`p_cam_transform` is already pointer of `godot_Transform`. i don't think we need to dereference it again.
```suggestion
	godot::Transform *cam_transform = (godot::Transform *)p_cam_transform;
```

Reply:
Thanks:)